### ISSUE TITLE
Match statement on a tuple of class/instance variables does not narrow their types in case bodies

### ISSUE THREAD

**Issue by bavalpey** [microsoft/pyright#8897]
**Describe the bug**
Pyright loses its ability to narrow the types of instance variables in match statements when they are placed into a tuple for multiple-matching.

Pyright is already able to narrow the type of instance variables that appear in case bodies (see the first example in the screenshot).
It is also able to narrow the type of normal variables that are matched using the tuple (see the second example in the screenshot).

It should therefore be able to narrow the type of instance variables that are matched in a tuple, but currently does not.

**Code or Screenshots**

```python
from typing import reveal_type

class Foo(): pass
class Bar(Foo): pass
class Baz():
    qux: Foo
    quux: Foo

def buggy(a: Baz):
    match (a.qux, a.quux):
        case (Bar(), Bar()):
            reveal_type(a.qux) # Should report Bar
            reveal_type(a.quux) # Should report Bar
```

![image](https://github.com/user-attachments/assets/1e9bc810-0f7a-44e1-a563-ce3408671785)

**VS Code extension or command-line**
The bug occurs on both the VS Code extension and the command line.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.380.
